include/zephyr/net/dsa_core.h

```diff
@@ -0,0 +1,274 @@
+/*
+ * Copyright (c) 2020 DENX Software Engineering GmbH
+ *               Lukasz Majewski <lukma@denx.de>
+ * Copyright 2025 NXP
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @file
+ * @brief Distributed Switch Architecture (DSA)
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_DSA_CORE_H_
+#define ZEPHYR_INCLUDE_NET_DSA_CORE_H_
+
+#include <errno.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/net/net_if.h>
+#include <zephyr/net/phy.h>
+#include <zephyr/drivers/pinctrl.h>
+
+/**
+ * @brief Distributed Switch Architecture (DSA)
+ * @defgroup dsa_core Distributed Switch Architecture (DSA)
+ * @since 2.5
+ * @version 0.8.0
+ * @ingroup networking
+ * @{
+ */
+
+/** @cond INTERNAL_HIDDEN */
+
+#define DSA_PORT_MAX_COUNT 8
```

Comment:
This should be a Kconfig or even better derived from the DT

Reply:
Right, I will change to derive from DT.
Thanks.